perceval/components/abstract_processor.py

```diff
@@ -448,6 +448,28 @@ def add_port(self, m, port: APort, location: PortLocation = PortLocation.IN_OUT)
             self._out_ports[port] = port_range
         return self
 
+    def remove_port(self, m, location: PortLocation = PortLocation.IN_OUT):
+        if location in (PortLocation.IN_OUT, PortLocation.INPUT):
```

Comment:
Could you factor **in** vs **out** port handling ? Both `if` are basically copy/paste of each other.